### PR TITLE
remove noauth footgun (WIP)

### PR DESCRIPTION
**Pull request by dominictarr**
remove noauth footgun. this makes it difficult to use noauth in an insecure way (say with any network connection)

https://github.com/ssbc/multiserver/issues/26

needs more tests

### CODE REVIEW

File: plugins/ws.js [hunk 1]
```diff
@@ -3,6 +3,28 @@ var URL = require('url')
 var pull = require('pull-stream/pull')
 var Map = require('pull-stream/throughs/map')
 
+function safe_origin (origin, address, port) {
+
+  //if the connection is not localhost, we shouldn't trust
+  //the origin header. So, use address instead of origin
+  //if origin not set, then it's definitely not a browser.
+  if(!(address === '::1' || address === '128.0.0.1') || origin == undefined)
```
Comment (cryptix) [inline plugins/ws.js]: 128? I guess it should be `127.0.0.1` right? Is this a resolved address? will `localhost` work as well?

File: plugins/noauth.js [hunk 2]
```diff
@@ -1,16 +1,21 @@
 var pull = require('pull-stream')
 
 module.exports = function (opts) {
+  opts = opts || {}
   return {
     name: 'noauth',
     create: function (_opts) {
       return function (stream, cb) {
+        if(opts.footgun !== true && stream.address !== opts.address)
+          return cb(new Error('unexpected client address:'+stream.address))
```
Comment (regular) [inline plugins/noauth.js]:
``` suggestion
          return cb(new Error('noauth transform not allowed for remote address '+stream.address))
```

**Comment by cryptix** [comment]
Yikes. I thought this was already merged..

**Comment by dominictarr** [comment]
I think this got stalled... this networking stuff is a massive pain, because it's being used lots of ways, and there are lots of possible configurations, and it's hard to test. That isn't anyone's fault... this is something we need to get right, but is quite draining to work on.

**Comment by christianbundy** [comment]
@dominictarr If it'd be helpful I'd be down to do a call and try to work through this. No rush on my end, just trying to help with old PRs that are still open. :heart:

**Comment by arj03** [comment]
Closing and reopening to see if tests pass.

**Comment by arj03** [comment]
Tests are hanging because they do not properly close. Also this is a breaking change and I'm also not sure if this is really the best way to do it. And the docs are not updated. 

I'm going to close this and do the 128 change out of this PR. If anyone has energy to finish this be my guest to open again and do that.

**Pull request closed without merging**